clippy_lints/src/loops/manual_flatten.rs

```diff
@@ -71,3 +81,18 @@ pub(super) fn check<'tcx>(
         });
     }
 }
+
+fn contains_nested_tuple_struct(pats: &[Pat<'_>]) -> bool {
+    for pat in pats {
+        match pat.kind {
+            PatKind::TupleStruct(_, _, _) => {
+                return true;
+            },
+            PatKind::Tuple(pats, _) => {
+                return contains_nested_tuple_struct(pats);
+            },
+            _ => {},
+        }
+    }
+    false
```

Comment:
This can be made more consise:

```suggestion
    pats.iter().any(|pat| {
        matches!(pat.kind, PatKind::TupleStruct(..))
            || matches!(pat.kind, PatKind::Tuple(pats, _) if contains_nested_tuple_struct(pats))
    })
```

Reply:
Thank you for the feedback.